File: integration-tests-java/src/main/java/io/micronaut/jimmer/java/it/controller/UserRoleResources.java
package io.micronaut.jimmer.java.it.controller;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.*;
import io.micronaut.jimmer.java.it.entity.UserRole;
import io.micronaut.jimmer.java.it.entity.dto.UserRoleSpecification;
import io.micronaut.jimmer.java.it.repository.UserRoleRepository;
import io.micronaut.jimmer.java.it.service.IUserRoleService;
import jakarta.transaction.Transactional;
import java.util.List;
import java.util.UUID;

@Controller("/userRoleResources")
public class UserRoleResources {

    private final IUserRoleService iUserRoleService;

    private final UserRoleRepository userRoleRepository;

    public UserRoleResources(
            IUserRoleService iUserRoleService, UserRoleRepository userRoleRepository) {
        this.iUserRoleService = iUserRoleService;
        this.userRoleRepository = userRoleRepository;
    }

    @Get("/userRoleFindById")
    public HttpResponse<UserRole> userRoleFindById(@QueryValue UUID id) {
        return HttpResponse.ok(iUserRoleService.findById(id));
    }

    @Put("/updateUserRoleById")
    @Transactional(rollbackOn = Exception.class)
    public HttpResponse<Void> updateUserRoleById(@QueryValue UUID id) {
        iUserRoleService.updateById(id);
        return HttpResponse.ok();
    }

    @Delete("/delete")
    @Transactional(rollbackOn = Exception.class)
    public HttpResponse<Void> delete(@QueryValue UUID id) {
        iUserRoleService.deleteById(id);
        return HttpResponse.ok();
    }

    @Get("/deleteReverseById")
    public HttpResponse<UserRole> deleteReverseById(@QueryValue UUID id) {
        return HttpResponse.ok(iUserRoleService.deleteReverseById(id));
    }

    @Get("/testUserRoleSpecification")
    public HttpResponse<List<UserRole>> testUserRoleSpecification(
            @RequestBean UserRoleSpecification userRoleSpecification) {
        return HttpResponse.ok(userRoleRepository.find(userRoleSpecification));
    }
}

File: integration-tests-kotlin/src/main/kotlin/io/micronaut/jimmer/kotlin/it/controller/UserRoleResources.kt
package io.micronaut.jimmer.kotlin.it.controller

import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Delete
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Put
import io.micronaut.http.annotation.QueryValue
import io.micronaut.http.annotation.RequestBean
import io.micronaut.jimmer.kotlin.it.entity.UserRole
import io.micronaut.jimmer.kotlin.it.entity.dto.UserRoleSpecification
import io.micronaut.jimmer.kotlin.it.repository.UserRoleRepository
import io.micronaut.jimmer.kotlin.it.service.IUserRoleService
import jakarta.transaction.Transactional
import java.util.UUID

@Controller("/userRoleResources")
open class UserRoleResources(
    private val iUserRoleService: IUserRoleService,
    private val userRoleRepository: UserRoleRepository,
) {
    @Get("/userRoleFindById")
    fun userRoleFindById(
        @QueryValue id: UUID,
    ): HttpResponse<UserRole> = HttpResponse.ok(iUserRoleService.findById(id))

    @Put("/updateUserRoleById")
    @Transactional(rollbackOn = [Exception::class])
    open fun updateUserRoleById(
        @QueryValue id: UUID,
    ): HttpResponse<Void> {
        iUserRoleService.updateById(id)
        return HttpResponse.ok()
    }

    @Delete("/delete")
    @Transactional(rollbackOn = [Exception::class])
    open fun delete(
        @QueryValue id: UUID,
    ): HttpResponse<Void> {
        iUserRoleService.deleteById(id)
        return HttpResponse.ok()
    }

    @Get("/deleteReverseById")
    fun deleteReverseById(
        @QueryValue id: UUID,
    ): HttpResponse<UserRole> = HttpResponse.ok(iUserRoleService.deleteReverseById(id))

    @Get("/testUserRoleSpecification")
    fun testUserRoleSpecification(
        @RequestBean userRoleSpecification: UserRoleSpecification,
    ): HttpResponse<List<UserRole>> = HttpResponse.ok(userRoleRepository.find(userRoleSpecification))
}
